eng/build.ps1

```diff
@@ -190,6 +190,11 @@ foreach ($argument in $PSBoundParameters.Keys)
   }
 }
 
+# Default the BuildTargetFramework to the latest netcorapp version
+if (!$framework) {
+  $arguments += " /p:BuildTargetFramework=net5.0"
```

Comment:
Why do we need to do this?  It will cause things to behave slightly differently when building from root (global property) vs leaf (default value from props)

Reply:
When building a leaf we now build all configurations by default. The TargetFramework SDK doesn't filter anymore based on the `BuildAllConfigurations` property but based on the `BuildTargetFramework` property. `BuildTargetFramework` isn't set to a default value anymore.

Reply:
Why aren't we setting it to a default anymore?

Reply:
As we now sequence all dependencies into the projects instead of doing an upfront depproj restore, we don't need to filter the `TargetFrameworks` to `$(NetCoreAppCurrent)` anymore and can build all configurations instead which is the same behavior as in a customer application (defined by the sdk).

Reply:
> When building a leaf we now build all configurations by default. 

I see, this makes sense.

Reply:
Can we instead set `BuildTargetFramework` in `Build.proj` which is the one that comes into play when we're not building a leaf project?

Reply:
That's a great suggestion, let me do that as it will mean that `dotnet build` will continue to work from the repo root.

Reply:
At some point we should move all global properties being passed in from build scripts into Build.proj.